commit 786a85ce80fd37a11096307247e68093950e75a4
Author: WanWizard<wanwizard@wanwizard.eu>
Date:   Thu May 7 17:08:08 2020 +0100

    fix attempts to write after the channel has been closed

diff --git a/src/twisted/web/http.py b/src/twisted/web/http.py
index cfd1cdf..fd17c51 100644
--- a/src/twisted/web/http.py
+++ b/src/twisted/web/http.py
@@ -1024,6 +1024,11 @@ class Request:
         if self.finished:
             raise RuntimeError('Request.write called on a request after '
                                'Request.finish was called.')
+
+	# bail out if we have no channel to write to
+	if self.channel is None:
+		return
+
         if not self.startedWriting:
             self.startedWriting = 1
             version = self.clientproto

